### PR TITLE
Addresses #468: Added Yahoo News

### PR DESCRIPTION
**Pull request by Remorax**
Addresses #468 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added Yahoo News
- Added tests for Yahoo News

Link: https://desolate-retreat-47972.herokuapp.com/

Please review 😄

### CODE REVIEW

File: app/scrapers/yahoo.py [hunk 1]
```diff
@@ -63,3 +65,28 @@ def parse_video_response(soup):
...
         return urls
+
+    @staticmethod
+    def parse_news_response(soup):
+        """ Parse response and returns the urls
+
+            Returns: urls (list)
+                    [[Tile1, url1], [Title2, url2], ...]
+        """
+        divClass = 'dd algo NewsArticle'
```
Comment (vaibhavsingh97) [inline app/scrapers/yahoo.py]: instead of declaring and passing it in context we can use directly

File: app/scrapers/yahoo.py [hunk 2]
```diff
@@ -63,3 +65,28 @@ def parse_video_response(soup):
...
+        urls = []
+        for div in soup.findAll('div', attrs={'class': divClass}):
+            link = div.find('a', attrs={'class': 'fz-m'})
```
Comment (vaibhavsingh97) [inline app/scrapers/yahoo.py]: `xpath` will reduce lines of code 👍

File: test/.cache/v/cache/lastfailed [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (vaibhavsingh97) [inline test/.cache/v/cache/lastfailed]: I think this file should not be there

**Comment by Remorax** [comment]
@vaibhavsingh97 I have made all the changes you requested, except for the `xpath` part. Inside the main loop, there are only 2-3 lines of code for extracting the element, the rest is regex to get the URL. I'm unsure how much `xpath` would reduce that number to, since 2-3 lines are always inevitable I guess and if that much would be worth breaking the uniformity, since `xpath` has not been used anywhere else.
That's why I didn't change that part. Hope that's fine 😄

**Comment by Remorax** [comment]
The commits have been squashed, the merge conflicts resolved and the Travis and Codacy errors have been fixed as well!
Please review @vaibhavsingh97 @ParthS007 @realslimshanky

**Comment by bhaveshAn** [comment]
Please check why your provided link is showing **No such app**

**Comment by Remorax** [comment]
@bhaveshAn I accidentally deleted the old app while trying to create space for new apps, i've updated the link please check it now 😄

**Pull request merged**